src/countervalues/react.js

```diff
@@ -0,0 +1,457 @@
+export type Polling = {
+  // completely wipe all countervalues
+  wipe: () => void,
+  // one shot poll function
+  // TODO: is there any usecases returning promise here?
+  // It's a bit tricky to return Promise with current impl
+  poll: () => void,
+  // start background polling
+  start: () => void,
+  // stop background polling
+  stop: () => void,
+  // true when the polling is in progress
+  pending: boolean,
+  // if the last polling failed, there will be an error
+  error: ?Error,
+};
+
+export type Props = {
+  children: React$Node,
+  userSettings: CountervaluesSettings,
+  // the time to wait before the first poll when app starts (allow things to render to not do all at boot time)
+  pollInitDelay?: number,
+  // the minimum time to wait before two automatic polls (then one that happen whatever network/appstate events)
+  autopollInterval?: number,
+  savedState?: CounterValuesStateRaw,
+};
+
+const CountervaluesPollingContext = createContext<Polling>({
+  wipe: () => {},
+  poll: () => {},
+  start: () => {},
+  stop: () => {},
+  pending: false,
+  error: null,
+});
+
+const CountervaluesContext = createContext<CounterValuesState>(initialState);
+
+export function Countervalues({
+  children,
+  userSettings,
+  pollInitDelay = 1 * 1000,
+  autopollInterval = 120 * 1000,
+  savedState,
+}: Props) {
+  const [{ state, pending, error }, dispatch] = useReducer<FetchState, Action>(
+    fetchReducer,
+    initialFetchState
+  );
+
+  // trigger poll but not every time poll callback is changed
+  const [triggerPoll, setTriggerPoll] = useState(false);
+
+  useEffect(() => {
+    if (pending || !triggerPoll) return;
+    dispatch({ type: "pending" });
+    loadCountervalues(state, userSettings).then(
+      (state) => {
+        if (!Object.keys(state.status).length) return;
+        dispatch({ type: "success", payload: state });
+        setTriggerPoll(false);
+      },
+      (error) => {
+        dispatch({ type: "error", payload: error });
+        setTriggerPoll(false);
+      }
+    );
+  }, [pending, state, userSettings, triggerPoll]);
+
+  useEffect(() => {
+    if (!savedState?.status || !Object.keys(savedState.status).length) return;
+    dispatch({
+      type: "setCounterValueState",
+      payload: importCountervalues(savedState, userSettings),
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [savedState]);
+
+  // trigger poll for the frist render and when userSettings is not exactly the same
+  const isFirstRender = useRef(true);
+  const userSettingsRef = useRef(userSettings);
+  useEffect(() => {
+    if (isFirstRender) {
+      isFirstRender.current = false;
+    } else {
+      if (
+        JSON.stringify(userSettings) === JSON.stringify(userSettingsRef.current)
+      )
+        return;
+    }
+    setTriggerPoll(true);
+    userSettingsRef.current = userSettings;
+  }, [userSettings]);
+
+  const [isPolling, setIsPolling] = useState(true);
```

Comment:
I don't understand the difference between `isPolling` and `pending`

Reply:
`isPolling` state is used to trigger polling effect where `pending` is the state.
https://github.com/LedgerHQ/ledger-live-common/blob/countervalues-poc/src/countervalues/react.js#L160-L161

Reply:
`triggerPollling` would be more appropriate name

Reply:
but there is already a triggerPoll 🤔

Reply:
oh, i've noticed that `triggerPolling` also exists.

Reply:
`triggerPolling` for one shot fetching and `isPolling` for continuous syncing.